modules/scylladb/scylladb.go

```diff
@@ -0,0 +1,144 @@
+package scylladb
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	"github.com/docker/go-connections/nat"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	port           = "9042/tcp"
+	shardAwarePort = "19042/tcp"
+)
+
+type Container struct {
+	testcontainers.Container
+}
+
+// WithConfigFile sets the YAML config file to be used for the cassandra container
+// It will also set the "configFile" parameter to the path of the config file
+// as a command line argument to the container.
+func WithConfigFile(configFile string) testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		cf := testcontainers.ContainerFile{
+			HostFilePath:      configFile,
```

Comment:
suggestion: we've been standardising on passing in readers to functional options like this as it makes them more flexible. Also correcting the description which looks like it was copied from cassandra.
```suggestion
// WithConfig sets the YAML config file to be used for the ScyllaDB container
func WithConfig(r io.Reader) testcontainers.CustomizeRequestOption {
	return func(req *testcontainers.GenericContainerRequest) error {
		cf := testcontainers.ContainerFile{
			Reader:      r,
```

Reply:
I used the Cassandra and a few other implementations to understand how Testcontainers works. Thanks for the suggestion!